tests/Browser/ResultsTest.php

```diff
@@ -24,6 +24,73 @@ public function test_shows_item_ids()
         });
     }
 
+    public function test_shows_message_for_non_existing_item_ids()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->visit(new ResultsPage('Q1|Q2'))
+                ->assertSee('No mismatches have been found for')
+                ->assertSeeLink('Q1')
+                ->assertSeeLink('Q2');
+        });
+    }
+
+    public function test_shows_tables_for_existing_items_and_message_for_non_existing_items()
+    {
+        $import = ImportMeta::factory()
+        ->for(User::factory()->uploader())
+        ->create();
+
+        Mismatch::factory(2)
+            ->for($import)
+            ->state(new Sequence(
+                [
+                    'statement_guid' => 'Q2$a2b48f1f-426d-91b3-1e0e-1d3c7b236bd0',
+                    'property_id' => 'P610',
+                    'wikidata_value' => 'Q513'
+                ],
+                [
+                    'statement_guid' => 'q111$fc6ebc3f-4ea3-3cc8-0f09-a5608474754c',
+                    'property_id' => 'P571',
+                    'wikidata_value' => '4540 million years BCE'
+                ]
+            ))
+            ->create();
+
+        $expected = [
+            [
+                'item_label' => 'Earth (Q2)',
+                'property_label' => 'highest point',
+                'wikidata_value' => 'Mount Everest'
+            ],
+            [
+                'item_label' => 'Mars (Q111)',
+                'property_label' => 'inception',
+                'wikidata_value' => '4540 million years BCE'
+            ]
+        ];
+
+        $mismatches = Mismatch::all();
+
+        $this->browse(function (Browser $browser) use ($mismatches, $expected) {
+            $idsQuery = $mismatches->implode('item_id', '|');
+            $idsWithNotFound = $idsQuery . '|Q1|Q666';
+            $browser->visit(new ResultsPage($idsWithNotFound));
+
+            foreach ($mismatches as $i => $mismatch) {
+                $browser->assertSeeLink($expected[$i]['item_label'])
+                    ->assertSeeLink($expected[$i]['property_label'])
+                    ->assertSeeLink($expected[$i]['wikidata_value'])
+                    ->assertSee($mismatch->external_value)
+                    ->assertSeeLink($mismatch->importMeta->user->username)
+                    ->assertSee($mismatch->importMeta->created_at->toDateString());
+            }
+
+            $browser->assertSee('No mismatches have been found for')
+                ->assertSeeLink('Q1')
+                ->assertSeeLink('Q666');
+        });
+    }
```

Comment:
This entire test basically copies the two tests that you already have: one for the notice message box and one for the correct result tables. Since these two are actually covered with all their details, I don't think that the same level of detail needs to be repeated. Instead we could simply test only for what we want to make sure: that both the message box and the results table are there in cases when mismatches for some items have been found, but not for others.

I think it can be done in a rather simple and easy to understand test:
```suggestion
    public function test_shows_table_for_existing_item_and_message_for_non_existing_item()
    {
        // add only one mismatch for Q1
        Mismatch::factory()
            ->for(ImportMeta::factory()->for(User::factory()->uploader()))
            ->state(['statement_guid' => 'Q1$a2b48f1f-426d-91b3-1e0e-1d3c7b236bd0'])
            ->create();

        $this->browse(function (Browser $browser) {
            // check results for Q1 and Q2
            $browser->visit(new ResultsPage('Q1|Q2'));

            $browser->assertSeeIn('.results', 'Q1');
            $browser->assertSeeIn('.wikit-Message--notice', 'Q2');
        });
    }
```

Reply:
Oh wow, yeah that's more succinct!